core/internal/features_test.go

```diff
@@ -729,11 +730,22 @@ func TestIntegration_FluxMonitor_Deviation(t *testing.T) {
 
 	config, cfgCleanup := cltest.NewConfig(t)
 	defer cfgCleanup()
-	app, appCleanup := cltest.NewApplicationWithConfigAndKey(t, config,
+	app, appCleanup := cltest.NewApplicationWithConfig(t, config,
 		eth.NewClientWith(rpcClient, gethClient),
 	)
 	defer appCleanup()
 
+	// Helps to avoid deadlocks caused by multiple tests simultaneously writing
+	// to eth_txes in different order and locking on nonce
```

Comment:
Let's elaborate on this comment a little more - specifically its that we have two tests (transactions because we're using txdb) which, for the same DefaultKeyAddress, try to:
Tx1
1.  Read the nonce from keys (locking on address, nonce)
2. Write to eth_txes
Tx 2:
1. Write to eth_txes (I guess updating the state of the tx as completed?)
2. Update nonce in keys  (blocked on lock from tx 1)

If every test/transaction has a different random nonce, then tx 2 doesn't have to wait for tx 1 to release the lock on that (address, nonce)

Reply:
We could also mention in the comment (or PR description) that an alternative preferable approach is to test scope the address (instead of using DefaultKeyAddress) but its more work and we chose this shortcut to unflake the tests ASAP.